saidump/saidump.cpp

```diff
@@ -399,7 +417,170 @@ void dumpGraph(const TableDump& td)
     std::cout << "}" << std::endl;
 }
 
-int main(int argc, char ** argv)
+/*
+ preprocess the input json file to make sure it's a valid json file.
+*/
+static int preProcessFile(std::string file_name)
+{
+    std::ifstream input_file(file_name);
+
+    if (!input_file.is_open())
+    {
+        std::cerr << "Failed to open the input file " << file_name << std::endl;
+        SWSS_LOG_ERROR("Failed to open the input file %s.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::end);     // Move to the end of the file
+    int64_t file_size = input_file.tellg(); // Get the current position
+
+    if (file_size >= RDB_FILE_MAX_SIZE)
+    {
+        std::cerr << "Get " << file_name.c_str() << "'s size failure or its size " << file_size << "  >= "<< RDB_FILE_MAX_SIZE / 1024 / 1024<<" MB." << std::endl;
+        SWSS_LOG_ERROR("Get %s's size failure or its size %ld  >= %ld MB.", file_name.c_str(), file_size, RDB_FILE_MAX_SIZE / 1024 / 1024);
+        // Close the input file
+        input_file.close();
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::beg);     // Move to the begin of the file
+
+    // Read the content of the input file into a string
+    std::string content((std::istreambuf_iterator<char>(input_file)),
+                   std::istreambuf_iterator<char>());
+    input_file.close();
+
+    content = regex_replace(content, std::regex("\\},\\{\\r"), ",");
+
+    //erase the 1st and last char.
+    if (content.length() >= 2 && content[0] == '[' && content[content.length()-1] == ']')
+    {
+        content.erase(0, 1);
+        content.erase(content.length() - 1);
+    }
+
+    std::ofstream outputFile(file_name);
+
+    if (!outputFile.is_open())
+    {
+        std::cerr << "Failed to open the output file " << file_name << std::endl;
+        SWSS_LOG_ERROR("Failed to open the output file %s.", file_name.c_str());
```

Comment:
since you are using this scheme a lot ,and basically repeat all the code i propose to add macro like this:
```c
#define SWSS_LOG_STDERR_ERROR(msg, ...) { fprintf(stderr, ##__VA_ARGS__); SWSS_LOG_ERROR(msg, ##__VA_ARGS__); }

and use it where is required to print std err
```

Reply:
Fix it.